avm-transpiler/src/transpile.rs

```diff
@@ -245,6 +250,56 @@ fn handle_foreign_call(
     }
 }
 
+/// Emit an AVM EMITNOTEHASH instruction
+/// (an emitNoteHash brillig foreign call was encountered)
+/// Adds the new instruction to the avm instructions list.
+fn emit_emit_note_hash(
```

Comment:
fine with separation but highlighting these functions are the same

Reply:
Good call, merged into one function!